osu.Game/Online/Multiplayer/Room.cs

```diff
@@ -124,15 +125,13 @@ public void CopyFrom(Room other)
                 var localItem = Playlist.FirstOrDefault(i => i.BeatmapID == item.BeatmapID);
 
                 if (localItem != null)
-                {
                     item.Beatmap.Value.Metadata = localItem.Beatmap.Value.Metadata;
-                }
             }
 
-            foreach (var removeableItem in Playlist.Except(other.Playlist).ToArray())
+            foreach (var removeableItem in Playlist.Except(other.Playlist, new PlaylistEqualityComparer()).ToArray())
```

Comment:
i'm thinking we just want to clear the playlist and re-add all, unless every item matches. this current logic is going to potentially change the order of the items incorrectly on an incoming change (can't happen in current scenarios, but may in the future).

i think i'd prefer to not use the equality comparer or `Except`, and just rewrite this whole section into a local linq implementation. and at that point, we can use `.Equals` and move the equality back into `PlaylistItem` itself (as `IEquatable<PlaylistItem>`).

Reply:
Fair call.